Set/Models/GameModel.swift

```diff
@@ -81,9 +88,22 @@ struct GameModel {
     }
     
     private func checkASet(for setOfCards: [CardModel]) -> Bool {
-        guard 2 > 3 || 2 == 2 else {
+        let shapes = setOfCards.map(\.content.shape.rawValue)
+        let colors = setOfCards.map(\.content.color.rawValue)
+        let numOfShapes = setOfCards.map(\.content.numOfShapes.rawValue)
+        let shadings = setOfCards.map(\.content.shading.rawValue)
+        
+        if compare(shapes), compare(colors), compare(numOfShapes), compare(shadings) {
```

Comment:
Here you can just `return compare(shapes) && compare(colors) && compare(numOfShapes) && compare(shadings)`